modules/data.atmosphere/R/download.AmerifluxLBL.R

```diff
@@ -14,50 +17,79 @@
+
+
+download.AmerifluxLBL <- function(sitename, outfolder, start_date, end_date,
+                                  overwrite = FALSE, verbose = FALSE, username = "pecan", method,
+                                  useremail = "@", data_product = "BASE-BADM", data_policy = "CCBY4.0", ...) {
   
+  # Initial set-ups for amerifluxr packages
+  # get start/end year code works on whole years only
   start_date <- as.POSIXlt(start_date, tz = "UTC")
   end_date <- as.POSIXlt(end_date, tz = "UTC")
   
   start_year <- lubridate::year(start_date)
   end_year <- lubridate::year(end_date)
   
+  
+  site <- sub(".* \\((.*)\\)", "\\1", sitename)
+  
+  
+  
+  
+  
   # make sure output folder exists
   if (!file.exists(outfolder)) {
     dir.create(outfolder, showWarnings = FALSE, recursive = TRUE)
   }
   
-  # need to query to get full file name #this is Ameriflux version
-  url <- "http://wile.lbl.gov:8080/AmeriFlux/DataDownload.svc/datafileURLs"
-  json_query <- paste0("{\"username\":\"", username, "\",\"siteList\":[\"", site, "\"],\"intendedUse\":\"Research - Land model/Earth system model\",\"description\":\"PEcAn download\"}")
-  result <- httr::POST(url, body = json_query, encode = "json", httr::add_headers(`Content-Type` = "application/json"))
-  link <- httr::content(result)
   
-  ftplink <- NULL
-  if (length(link$dataURLsList) > 0) {
-    ftplink <- link$dataURLsList[[1]]$URL
+
+  repeat {
+    tout <- options("timeout")
+    zip_file <- try(amerifluxr::amf_download_base(user_id = username,
+                                                  user_email = useremail,
+                                                  site_id = site,
+                                                  data_product = data_product,
+                                                  data_policy = data_policy,
+                                                  agree_policy = TRUE,
+                                                  intended_use = "model",
+                                                  intended_use_text = "PEcAn download",
+                                                  verbose = verbose,
+                                                  out_dir = outfolder)
+    )
+    if (!inherits(zip_file, "try-error")){
+      break
+    }else if(tout$timeout > 250 ){
+      PEcAn.logger::logger.info("Download takes too long, check your connection.")
+      break
+    }
+    PEcAn.logger::logger.info("Added 100 seconds before the download timeouts")
+    options(timeout = tout$timeout + 100)
   }
   
-  # test to see that we got back a FTP
-  if (is.null(ftplink)) {
-    PEcAn.logger::logger.severe("Could not get information about", site, ".", "Is this an AmerifluxLBL site?")
+  
+  
+  
+  # Path to created zip-file
+  ftplink <- zip_file
+  if(!grepl(".zip", ftplink)){
+    PEcAn.logger::logger.info("Not able to download a zip-file. Check download.AmerifluxLBL inputs")
```

Comment:
likewise maybe we should change `logger.info` to `logger.severe` here although it will probably break and stop on its own